### PR TITLE
DataFrame.to_csv()

### PR DESCRIPTION
**Pull request by RainFung**
How to do unit test and doc test in to_xxx

### CODE REVIEW

**Comment by RainFung** [comment]
Meet too many unsolved problem like unitest and py-lint .Need help to finish the rest work.

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -2191,6 +2191,111 @@ def to_parquet(self, path: str, mode: str = 'error',
...
+        ...    code=[1, 2 ,3]), columns=['date', 'country', 'code'])
+        >>> df
+                         date country  code
+        0 2012-01-31 12:00:00      KR     1
+        1 2012-02-29 12:00:00      US     2
+        2 2012-03-31 12:00:00      JP     3
+
+        >>> df.to_csv(path=r'%s/to_csv/foo.csv' % path)
+        """
+        self._sdf.coalesce(1).write.csv(path, mode=mode, compression=compression, sep=sep,
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply databricks/koalas/frame.py]: Yea .. let's don't do `coalesce` at all for now actually to be consistent with other APIs like `to_parquet` and `to_delta` for now.

[the remark this answers is not in the document]
Comment (RainFung) [reply databricks/koalas/frame.py]: remove `coalesce ` already. But i cant pass unitest....

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -2191,6 +2191,111 @@ def to_parquet(self, path: str, mode: str = 'error',
...
+        sep: str, default None
+            Sets a single character as a separator for each field and value. If None is set,
+            it uses the default value, ``,``.
+        quote: str, default None
+            Sets a single character used for escaping quoted values where the separator can be part
+            of the value. If None is set, it uses the default value, ``"``.
+            If an empty string is set, it uses ``u0000`` (null character).
+        escape: str, default None
+            Sets a single character used for escaping quotes inside an already quoted value.
+            If None is set, it uses the default value, ``\``.
```
Comment (garawalid) [inline databricks/koalas/frame.py]: @RainFung 
This ``\`` is the root of the problem. 
Pydocstyle checks passed when I add `r` at the beginning of the docstring. 
```python 
r"""
Saves the content of the ...
```

Please make sure that the documentation still rendering correctly.

**Comment by HyukjinKwon** [comment]
@RainFung, you don't have to make all the tests pass for now because Koalas is still being rapidly evolved. Skip or remove the tests.

I think the problem is that `Series.to_csv` is disappeared by this PR.

**Comment by ueshin** [comment]
I might miss the context, but seems like the arguments are changing to follow Spark's method. Should we follow pandas' as far as possible?

**Comment by HyukjinKwon** [comment]
Yes, we should follow Pandas' as possible as we can.

**Comment by HyukjinKwon** [comment]
gentle ping @RainFung to update.

**Comment by RainFung** [comment]
I'll update it today.

**Comment by HyukjinKwon** [comment]
@RainFung, you can address https://github.com/databricks/koalas/issues/524#issuecomment-518364596 together here.

**Comment by HyukjinKwon** [comment]
@RainFung, can you update this one please? CSV is pretty common API and I think we should match and fix ASAP.

**Comment by RainFung** [comment]
Ok, i'll create some PR like to_xxx recently.

**Pull request closed without merging**